File: leetcode_src/src/ContainsDuplicate.java
// https://leetcode.com/problems/contains-duplicate/

import java.util.*;

public class ContainsDuplicate {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> s = new HashSet<>();
        for(int n : nums) {
            if(!s.add(n)) {
                return true;
            }
        }
        return false;
    }

    public boolean containsDuplicate2(int[] nums)
    {
        int min = Integer.MAX_VALUE;
        int max = Integer.MIN_VALUE;
        for(int i = 0; i < nums.length; i++) {
            if(nums[i] < min) min = nums[i];
            if(nums[i] > max) max = nums[i];
        }
        boolean[] visited = new boolean[max - min + 1];
        for(int i = 0; i < nums.length; i++) {
            int idx = nums[i] - min;
            if(visited[idx] == true) return true;
            visited[idx] = true;
        }
        return false;
    }
}

File: leetcode_src/src/ContainsDuplicate.py
from typing import *

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        s1 = set()

        for n in nums:
            if n in s1:
                return True
            else:
                s1.add(n)

        return False

    def containsDuplicate2(self, nums: List[int]) -> bool:
        return len(nums) != len(set(nums))

print(Solution().containsDuplicate2([1, 2, 3, 1]))
